server/http.go

```diff
@@ -581,54 +572,15 @@ func (p *Plugin) handleStartMeeting(w http.ResponseWriter, r *http.Request) {
 		topic = defaultMeetingTopic
 	}
 
-	var meetingID int
-	var createMeetingErr error
-	userPMISettingPref, err := p.getPMISettingData(user.Id)
+	meetingID, err := p.handleMeetingCreation(req.ChannelID, req.RootID, topic, user, zoomUser, w)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
-		p.askPreferenceForMeeting(user.Id, req.ChannelID, req.RootID)
-		return
-	}
-
-	createMeetingWithPMI := false
-	switch userPMISettingPref {
-	case "", zoomPMISettingValueAsk:
-		p.askPreferenceForMeeting(user.Id, req.ChannelID, req.RootID)
-
-		if _, err = w.Write([]byte(`{"meeting_url": ""}`)); err != nil {
-			p.API.LogWarn("failed to write the response", "Error", err.Error())
-		}
-		return
-	case trueString:
-		createMeetingWithPMI = true
-		meetingID = zoomUser.Pmi
-
-		if meetingID <= 0 {
-			meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
-			if createMeetingErr != nil {
-				p.API.LogWarn("failed to create the meeting", "Error", createMeetingErr.Error())
-				http.Error(w, createMeetingErr.Error(), http.StatusInternalServerError)
-				return
-			}
-			p.sendEnableZoomPMISettingMessage(userID, req.ChannelID, req.RootID)
-		}
-	default:
-		meetingID, createMeetingErr = p.createMeetingWithoutPMI(user, zoomUser, req.ChannelID, topic)
-		if createMeetingErr != nil {
-			p.API.LogWarn("failed to create the meeting", "Error", createMeetingErr.Error())
-			http.Error(w, createMeetingErr.Error(), http.StatusInternalServerError)
-			return
-		}
 	}
 
-	if err = p.postMeeting(user, meetingID, req.ChannelID, req.RootID, topic); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+	if meetingID == 0 {
```

Comment:
Can we return the meeting URL from the function instead of ID. In that case we will not have to add this conditional and we will not need to pass the io.writer to `handleMeetingCreation`